File: ImperatorToCK3Tests/AccessoryGenesTests.cpp
#include "gtest/gtest.h"
#include "Imperator/Genes/AccessoryGenes.h"
#include <sstream>



TEST(ImperatorWorld_AccessoryGenesTests, indexCanBeSet) {
	std::stringstream input;
	input << "=\n";
	input << "{\n";
	input << "\tindex=69";
	input << "}";

	const Imperator::AccessoryGenes theGenes(input);

	ASSERT_EQ(69, theGenes.getIndex());
}


TEST(ImperatorWorld_AccessoryGenesTests, indexDefaultsTo0) {
	std::stringstream input;
	input << "=\n";
	input << "{\n";
	input << "}";

	const Imperator::AccessoryGenes theGenes(input);

	ASSERT_EQ(0, theGenes.getIndex());
}

TEST(ImperatorWorld_AccessoryGenesTests, genesDefaultToEmpty) {
	std::stringstream input;
	input << "=\n";
	input << "{\n";
	input << "}";

	const Imperator::AccessoryGenes theGenes(input);

	ASSERT_TRUE(theGenes.getGenes().empty());
}

TEST(ImperatorWorld_AccessoryGenesTests, accessoryGenesAreProperlyLoaded) {
	std::stringstream input;
	input << "= {\n";
	input << "index= 5\n";
	input << "\thairstyles = {\n";
	input << "\t\tindex = 1\n";
	input << "\t}\n";
	input << "\tclothes = {\n";
	input << "\t\tindex = 2\n";
	input << "\t}\n";
	input << "}";


	const Imperator::AccessoryGenes theGenes(input);
	ASSERT_EQ(5, theGenes.getIndex());
	ASSERT_EQ(2, theGenes.getGenes().size());
	ASSERT_EQ(1, theGenes.getGenes().find("hairstyles")->second.getIndex());
	ASSERT_EQ(2, theGenes.getGenes().find("clothes")->second.getIndex());
}

File: ImperatorToCK3.UnitTests/CommonUtils/Genes/AccessoryGenesTests.cs
using commonItems;
using ImperatorToCK3.CommonUtils.Genes;
using Xunit;

namespace ImperatorToCK3.UnitTests.CommonUtils.Genes {
	[Collection("Sequential")]
	[CollectionDefinition("Sequential", DisableParallelization = true)]
	public class AccessoryGenesTests {
		[Fact]
		public void IndexCanBeSet() {
			var reader = new BufferedReader(
				"=\n" +
				"{\n" +
				"\tindex=69" +
				"}"
			);
			var genes = new AccessoryGenes(reader);

			Assert.Equal((uint)69, genes.Index);
		}

		[Fact]
		public void IndexDefaultsTo0() {
			var reader = new BufferedReader("= {}");
			var genes = new AccessoryGenes(reader);

			Assert.Equal((uint)0, genes.Index);
		}

		[Fact]
		public void GenesDefaultToEmpty() {
			var reader = new BufferedReader("= {}");
			var genes = new AccessoryGenes(reader);

			Assert.Empty(genes.Genes);
		}

		[Fact]
		public void AccessoryGenesAreProperlyLoaded() {
			var reader = new BufferedReader(
				"= {\n" +
				"index= 5\n" +
				"\thairstyles = {\n" +
				"\t\tindex = 1\n" +
				"\t}\n" +
				"\tclothes = {\n" +
				"\t\tindex = 2\n" +
				"\t}\n" +
				"}"
			);
			var genes = new AccessoryGenes(reader);
			Assert.Equal((uint)5, genes.Index);
			Assert.Equal(2, genes.Genes.Count);
			Assert.Equal((uint)1, genes.Genes["hairstyles"].Index);
			Assert.Equal((uint)2, genes.Genes["clothes"].Index);
		}
	}
}
